README.md
# Sucharu Icons theme

Freedom-3D Icons is a [Material Design](https://material.io) theme for GNOME/GTK based desktop environments.

Based on XONE II Icons --  [XONE_II_ICON_THEME](https://github.com/unc926/XONE_II_ICON_THEME)

Based on korla Icons -- [korla Icon theme](https://github.com/bikass/korla)

## Installation
To install themes :

    1) downdload zip file
    2) copy all the themes and pest it in /home/{ put your username }/.icons directory
    3) apply themes and enjoy 😀😀😀

### Preview

![MacOS-3D](preview/1.png?raw=true)
![MacOS-3D](preview/2.png?raw=true)
![MacOS-3D](preview/3.png?raw=true)
![MacOS-3D](preview/4.png?raw=true)
![MacOS-3D](preview/5.png?raw=true)

## Donate


debian/changelog
mamolinux-icons (1.0.2ubuntu1) stable; urgency=medium

  * Add Templates and bitmaps rendering code from Yaru
  * Add workflow to update mamolinux/stable branch
  * Fix conflict with Sucharu new icon theme (#4)
    - Rename Sucharu to Sucharu-Alt
    - Rename Sucharu-Dark to Sucharu-Dark-Alt

 -- Himadri Sekhar Basu <hsb10@iitbbs.ac.in>  Mon, 13 Mar 2023 19:56:48 +0530

mamolinux-icons (1.0.2) stable; urgency=medium

  * Replace symlinks with actual directories
    - Use separate files and directory for Aqua and Dark-Aqua
    - Use separate files and directory for Sand and Dark-Sand
    - Use separate files and directory for Teal and Dark-Teal
  * Update debian packaging and lintian cleanup
    - Add misc:Depends and Homepage in control
    - Update standards version in control
    - Update upstream contacts and copyright in copyright
    - Add CC-BY-SA paragraph in copyright
    - Remove postinst and postrm scripts as they are automatically generated
    - Remove builddeb override for compression in rules
  * Fix error while cache generation and typos

 -- Himadri Sekhar Basu <hsb10@iitbbs.ac.in>  Sun, 02 Jan 2022 00:49:14 +0530

mamolinux-icons (1.0.1) stable; urgency=medium

  * Update debian packaging
  * Rename theme from freedom to sucharu
  * Add fullcolour actions from Yaru and Mint-Y
  * Add source svg for application icons in apps
  * Add source svg for category icons in categories
  * Add source svg for devices icons in devices
  * Add source svg for mimetypes icons in mimetypes
  * Add source svg for places icons in places
  * Add source svg for status icons in status
  * Add list of icons and symlinks and renderer files
  * Update debian packaging and readme
  * Directly put all icons in usr instead of building
    - Use Mint colour as default variant
    - Provide blue as separate variant
    - Provide Blue as Aqua
    - Provide Brown
    - Provide Green
    - Provide Grey
    - Provide Orange
    - Provide Pink
    - Provide Purple
    - Provide Red
    - Provide Orange as Sand
    - Provide Green as Teal
    - Provide Yellow

 -- Himadri Sekhar Basu <hsb10@iitbbs.ac.in>  Sat, 01 Jan 2022 20:45:29 +0530

mamolinux-icons (1.0.0) stable; urgency=medium

  * Initial release. (Closes: #XXXXXX)

 -- Himadri Sekhar Basu <hsb10@iitbbs.ac.in>  Sun, 28 Nov 2021 19:13:40 +0530


src/render_devices.sh
#!/bin/bash

INKSCAPE="/usr/bin/inkscape"
OPTIPNG="/usr/bin/optipng"

pushd `dirname $0` > /dev/null
DIR="$( cd "$(dirname "$0")" ; pwd -P )"
popd > /dev/null

ulimit -s 1024

cd ${DIR}

THEMEDIR=../usr/share/icons/Sucharu

mkdir -p $THEMEDIR

for CONTEXT in devices
do

    mkdir -p $THEMEDIR/$CONTEXT
    mkdir -p $THEMEDIR/$CONTEXT

    cp -r $CONTEXT/symbolic $THEMEDIR/$CONTEXT

    for SIZE in 16 22 24 32 48 64 96 128
    do
        $INKSCAPE -S $CONTEXT.svg | grep -E "_$SIZE" | sed 's/\,.*$//' > index.tmp

        mkdir -p $THEMEDIR/$CONTEXT/$SIZE
        mkdir -p $THEMEDIR/$CONTEXT/$SIZE@2x

        cp -r $CONTEXT/symlinks/* $THEMEDIR/$CONTEXT/$SIZE
        cp -r $CONTEXT/symlinks/* $THEMEDIR/$CONTEXT/$SIZE@2x

        for OBJECT_ID in `cat index.tmp`
        do

            ICON_NAME=$(sed "s/\_$SIZE.*$//" <<< $OBJECT_ID)

            if [ -f $THEMEDIR/$CONTEXT/$SIZE/$ICON_NAME.png ]; then
                echo $THEMEDIR/$CONTEXT/$SIZE/$ICON_NAME.png exists. Removing ...
                rm -f $THEMEDIR/$CONTEXT/$SIZE/$ICON_NAME.png
                echo Rendering $THEMEDIR/$CONTEXT/$SIZE/$ICON_NAME.png
                $INKSCAPE --export-id=$OBJECT_ID \
                          --export-id-only \
                          --export-png=$THEMEDIR/$CONTEXT/$SIZE/$ICON_NAME.png $CONTEXT.svg >/dev/null \
                && $OPTIPNG -o7 --quiet $ASSETS_DIR/$i.png
            else
                echo
                echo Rendering $THEMEDIR/$CONTEXT/$SIZE/$ICON_NAME.png
                $INKSCAPE --export-id=$OBJECT_ID \
                          --export-id-only \
                          --export-png=$THEMEDIR/$CONTEXT/$SIZE/$ICON_NAME.png $CONTEXT.svg >/dev/null \
                && $OPTIPNG -o7 --quiet $ASSETS_DIR/$i.png
            fi
            if [ -f $THEMEDIR/$CONTEXT/$SIZE@2x/$ICON_NAME.png ]; then
                echo $THEMEDIR/$CONTEXT/$SIZE@2x/$ICON_NAME.png exists. Removing ...
                rm -f $THEMEDIR/$CONTEXT/$SIZE@2x/$ICON_NAME.png
                echo Rendering $THEMEDIR/$CONTEXT/$SIZE@2x/$ICON_NAME.png
                $INKSCAPE --export-id=$OBJECT_ID \
                          --export-dpi=180 \
                          --export-id-only \
                          --export-png=$THEMEDIR/$CONTEXT/$SIZE@2x/$ICON_NAME.png $CONTEXT.svg >/dev/null \
                && $OPTIPNG -o7 --quiet $ASSETS_DIR/$i@2.png 
            else
                echo
                echo Rendering $THEMEDIR/$CONTEXT/$SIZE@2x/$ICON_NAME.png
                $INKSCAPE --export-id=$OBJECT_ID \
                          --export-dpi=180 \
                          --export-id-only \
                          --export-png=$THEMEDIR/$CONTEXT/$SIZE@2x/$ICON_NAME.png $CONTEXT.svg >/dev/null \
                && $OPTIPNG -o7 --quiet $ASSETS_DIR/$i@2.png 
            fi
        done
    done
done

# rm index.tmp
# cp index.theme $THEMEDIR/index.theme
# rm -rf $THEMEDIR/actions/{32,32@2x,48,48@2x,64,64@2x,96,96@2x,128,128@2x} # derp

# TODO
# cp -r animations $THEMEDIR/.
# cp -r panel $THEMEDIR/.



src/render-devices-dir.py
#!/usr/bin/python3

import os

for size in [16, 22, 24, 32, 48, 64, 96, 128]:
    source_path = "devices/%s" % size
    dest_path = "../usr/share/icons/Sucharu/devices/%s" % size
    dest2x_path = "../usr/share/icons/Sucharu/devices/%s@2x" % size
    os.system("mkdir -p %s %s" % (dest_path, dest2x_path))
    
    for filename in sorted(os.listdir(source_path)):
        png_filename = filename.replace(".svg", ".png")
        source_file = os.path.join(source_path, filename)
        dest_file = os.path.join(dest_path, png_filename)
        dest2x_file = os.path.join(dest2x_path, png_filename)
        
        if not os.path.exists(dest_file):
            print ("Rendering %s" % dest_file)
            os.system("inkscape --export-png=%s -f %s >/dev/null && optipng -o7 --quiet %s" % (dest_file, source_file, dest_file))
        
        if not os.path.exists(dest2x_file):
            print ("Rendering %s" % dest2x_file)
            os.system("inkscape --export-dpi=192 --export-png=%s -f %s >/dev/null && optipng -o7 --quiet %s" % (dest2x_file, source_file, dest2x_file))
